### PR TITLE
Phix

### PR DESCRIPTION
**Pull request by petelomax**
CHANGES and SUPPORTED_LANGUAGES for Phix

https://github.com/highlightjs/highlight.js/issues/3727

### Changes
Minimal

### Checklist
- [X] Added markup tests, or they don't apply here because...
- [X] Updated the changelog at `CHANGES.md`

### CODE REVIEW

**Comment by joshgoebel** [comment]
Broken link?  Do you have the repo elsewhere and were wanting to move it here?

**Comment by petelomax** [comment]
Yes, my copy is at https://github.com/petelomax/highlightjs-phix

**Comment by joshgoebel** [comment]
Could you please add a `dist` folder with the CDN build artifacts?

**Comment by petelomax** [comment]
I could probably add an empty dist folder, but I have absolutely no idea what "CDN build artifacts" are.

**Comment by joshgoebel** [comment]
> but I have absolutely no idea what "CDN build artifacts" are.

The files that `build cdn` drops into your repo if you're using the blessed development structure.  The 3rd party developer document should explain how all that works.

https://github.com/highlightjs/highlight.js/blob/main/extra/3RD_PARTY_QUICK_START.md

**Comment by RobloxianDemo** [comment]
> I could probably add an empty dist folder, but I have absolutely no idea what "CDN build artifacts" are.

I already created a PR for you [found here](https://github.com/highlightjs/highlightjs-phix/pull/1), which has already sorted this for you, so feel free to merge it anytime.

**Comment by petelomax** [comment]
What's happening on this?

**Comment by joshgoebel** [comment]
The issue is still:

https://github.com/highlightjs/highlight.js/pull/3728#issuecomment-1806644823

You need to use our build process to create the distributables (as well as scan for security issues)...  it's all documented in the docs I link to.  The build system creates the `dist` folder, then you add it to your github repo.

**Comment by petelomax** [comment]
node ./tools/build.js -t node
Starting build.
Finished build.
(node:5646) UnhandledPromiseRejectionWarning: /home/pete/highlight.js/node_modules/rollup/dist/shared/rollup.js:58
            textEncoder ??= new TextEncoder();
                         ^

SyntaxError: Unexpected token '?'
    at wrapSafe (internal/modules/cjs/loader.js:915:16)
    at Module._compile (internal/modules/cjs/loader.js:963:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Module.require (internal/modules/cjs/loader.js:887:19)
    at require (internal/modules/cjs/helpers.js:74:18)
    at Module.<anonymous> (/home/pete/highlight.js/node_modules/rollup/dist/rollup.js:14:16)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
(node:5646) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:5646) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

**Comment by joshgoebel** [comment]
Are you using some very old version of Node.js?

**Comment by petelomax** [comment]
node -v says v12.22.9

**Comment by petelomax** [comment]
however, node ./tools/build.js -t cdn
Starting build.
Finished build.
Writing style files.
Building language files.
..................................................................................................................................................................................................
Building extra/highlightjs-phix/dist/phix.min.js.

Building highlight.js.
Error [RollupError]: "default" is not exported by "src/languages/phix.js", imported by "builtInLanguages".
    at Object.error (/home/pete/highlight.js/node_modules/rollup/dist/shared/parseAst.js:279:30)
    at Module.error (/home/pete/highlight.js/node_modules/rollup/dist/shared/rollup.js:13885:28)
    at Module.getVariableForExportName (/home/pete/highlight.js/node_modules/rollup/dist/shared/rollup.js:14049:29)
    at Module.includeAllExports (/home/pete/highlight.js/node_modules/rollup/dist/shared/rollup.js:14122:37)
    at Object.includeAllExports (/home/pete/highlight.js/node_modules/rollup/dist/shared/rollup.js:14251:43)
    at NamespaceVariable.include (/home/pete/highlight.js/node_modules/rollup/dist/shared/rollup.js:13324:22)
    at Module.includeVariable (/home/pete/highlight.js/node_modules/rollup/dist/shared/rollup.js:14591:22)
    at Module.includeVariableInModule (/home/pete/highlight.js/node_modules/rollup/dist/shared/rollup.js:14609:14)
    at Identifier.include (/home/pete/highlight.js/node_modules/rollup/dist/shared/rollup.js:8557:30)
    at MemberExpression.includeCallArguments (/home/pete/highlight.js/node_modules/rollup/dist/shared/rollup.js:3299:22) {
  binding: 'default',
  code: 'MISSING_EXPORT',
  exporter: '/home/pete/highlight.js/src/languages/phix.js',
  id: 'builtInLanguages',
  url: 'https://rollupjs.org/troubleshooting/#error-name-is-not-exported-by-module',
  pos: 8775,
  loc: { column: 9, file: 'builtInLanguages', line: 131 },
  frame: '129: export { default as grmr_pf } from "./src/languages/pf.js";\n' +
    '130: export { default as grmr_pgsql } from "./src/languages/pgsql.js";\n' +
    '131: export { default as grmr_phix } from "./src/languages/phix.js";\n' +
    '              ^\n' +
    '132: export { default as grmr_php } from "./src/languages/php.js";\n' +
    '133: export { default as grmr_php_template } from "./src/languages/php-template.js";',
  watchFiles: [
    '/home/pete/highlight.js/src/stub.js',
    '/home/pete/highlight.js/src/highlight.js',
    '/home/pete/highlight.js/src/languages/1c.js',
    '/home/pete/highlight.js/src/languages/abnf.js',
    '/home/pete/highlight.js/src/languages/accesslog.js',
    '/home/pete/highlight.js/src/languages/actionscript.js',
    '/home/pete/highlight.js/src/languages/ada.js',
    '/home/pete/highlight.js/src/languages/angelscript.js',
    '/home/pete/highlight.js/src/languages/apache.js',
    '/home/pete/highlight.js/src/languages/applescript.js',
    '/home/pete/highlight.js/src/languages/arcade.js',
    '/home/pete/highlight.js/src/languages/armasm.js',
    '/home/pete/highlight.js/src/languages/arduino.js',
    '/home/pete/highlight.js/src/languages/xml.js',
    '/home/pete/highlight.js/src/languages/asciidoc.js',
    '/home/pete/highlight.js/src/languages/aspectj.js',
    '/home/pete/highlight.js/src/languages/autohotkey.js',
    '/home/pete/highlight.js/src/languages/autoit.js',
    '/home/pete/highlight.js/src/languages/avrasm.js',
    '/home/pete/highlight.js/src/languages/awk.js',
    '/home/pete/highlight.js/src/languages/axapta.js',
    '/home/pete/highlight.js/src/languages/bash.js',
    '/home/pete/highlight.js/src/languages/bnf.js',
    '/home/pete/highlight.js/src/languages/basic.js',
    '/home/pete/highlight.js/src/languages/brainfuck.js',
    '/home/pete/highlight.js/src/languages/c.js',
    '/home/pete/highlight.js/src/languages/cal.js',
    '/home/pete/highlight.js/src/languages/capnproto.js',
    '/home/pete/highlight.js/src/languages/ceylon.js',
    '/home/pete/highlight.js/src/languages/clean.js',
    '/home/pete/highlight.js/src/languages/clojure.js',
    '/home/pete/highlight.js/src/languages/clojure-repl.js',
    '/home/pete/highlight.js/src/languages/cmake.js',
    '/home/pete/highlight.js/src/languages/coffeescript.js',
    '/home/pete/highlight.js/src/languages/coq.js',
    '/home/pete/highlight.js/src/languages/cos.js',
    '/home/pete/highlight.js/src/languages/cpp.js',
    '/home/pete/highlight.js/src/languages/crmsh.js',
    '/home/pete/highlight.js/src/languages/crystal.js',
    '/home/pete/highlight.js/src/languages/csharp.js',
    '/home/pete/highlight.js/src/languages/csp.js',
    '/home/pete/highlight.js/src/languages/css.js',
    '/home/pete/highlight.js/src/languages/d.js',
    '/home/pete/highlight.js/src/languages/markdown.js',
    '/home/pete/highlight.js/src/languages/dart.js',
    '/home/pete/highlight.js/src/languages/delphi.js',
    '/home/pete/highlight.js/src/languages/diff.js',
    '/home/pete/highlight.js/src/languages/django.js',
    '/home/pete/highlight.js/src/languages/dns.js',
    '/home/pete/highlight.js/src/languages/dockerfile.js',
    '/home/pete/highlight.js/src/languages/dos.js',
    '/home/pete/highlight.js/src/languages/dsconfig.js',
    '/home/pete/highlight.js/src/languages/dts.js',
    '/home/pete/highlight.js/src/languages/dust.js',
    '/home/pete/highlight.js/src/languages/ebnf.js',
    '/home/pete/highlight.js/src/languages/elixir.js',
    '/home/pete/highlight.js/src/languages/elm.js',
    '/home/pete/highlight.js/src/languages/ruby.js',
    '/home/pete/highlight.js/src/languages/erb.js',
    '/home/pete/highlight.js/src/languages/erlang-repl.js',
    '/home/pete/highlight.js/src/languages/erlang.js',
    '/home/pete/highlight.js/src/languages/excel.js',
    '/home/pete/highlight.js/src/languages/fix.js',
    '/home/pete/highlight.js/src/languages/flix.js',
    '/home/pete/highlight.js/src/languages/fortran.js',
    '/home/pete/highlight.js/src/languages/fsharp.js',
    '/home/pete/highlight.js/src/languages/gams.js',
    '/home/pete/highlight.js/src/languages/gauss.js',
    '/home/pete/highlight.js/src/languages/gcode.js',
    '/home/pete/highlight.js/src/languages/gherkin.js',
    '/home/pete/highlight.js/src/languages/glsl.js',
    '/home/pete/highlight.js/src/languages/gml.js',
    '/home/pete/highlight.js/src/languages/go.js',
    '/home/pete/highlight.js/src/languages/golo.js',
    '/home/pete/highlight.js/src/languages/gradle.js',
    '/home/pete/highlight.js/src/languages/graphql.js',
    '/home/pete/highlight.js/src/languages/groovy.js',
    '/home/pete/highlight.js/src/languages/haml.js',
    '/home/pete/highlight.js/src/languages/handlebars.js',
    '/home/pete/highlight.js/src/languages/haskell.js',
    '/home/pete/highlight.js/src/languages/haxe.js',
    '/home/pete/highlight.js/src/languages/hsp.js',
    '/home/pete/highlight.js/src/languages/http.js',
    '/home/pete/highlight.js/src/languages/hy.js',
    '/home/pete/highlight.js/src/languages/inform7.js',
    '/home/pete/highlight.js/src/languages/ini.js',
    '/home/pete/highlight.js/src/languages/irpf90.js',
    '/home/pete/highlight.js/src/languages/isbl.js',
    '/home/pete/highlight.js/src/languages/java.js',
    '/home/pete/highlight.js/src/languages/javascript.js',
    '/home/pete/highlight.js/src/languages/jboss-cli.js',
    '/home/pete/highlight.js/src/languages/json.js',
    '/home/pete/highlight.js/src/languages/julia.js',
    '/home/pete/highlight.js/src/languages/julia-repl.js',
    '/home/pete/highlight.js/src/languages/kotlin.js',
    '/home/pete/highlight.js/src/languages/lasso.js',
    '/home/pete/highlight.js/src/languages/latex.js',
    '/home/pete/highlight.js/src/languages/ldif.js',
    '/home/pete/highlight.js/src/languages/leaf.js',
    '/home/pete/highlight.js/src/languages/less.js',
    ... 115 more items
  ]
}
/home/pete/highlight.js/tools/lib/bundling.js:11
  const { output } = await bundle.generate(outputOptions);
                                  ^

TypeError: Cannot read properties of undefined (reading 'generate')
    at generate (/home/pete/highlight.js/tools/lib/bundling.js:11:35)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async rollupCode (/home/pete/highlight.js/tools/lib/bundling.js:5:18)
    at async buildCore (/home/pete/highlight.js/tools/build_browser.js:231:17)
    at async Object.buildCDN [as build] (/home/pete/highlight.js/tools/build_cdn.js:56:16)
    at async doTarget (/home/pete/highlight.js/tools/build.js:90:3)

Node.js v20.12.2

Same error whether I had last year's function hljsDefinePhix(hljs) { or this year's export default function(hljs) {.

**Comment by petelomax** [comment]
There is now a dist file in https://github.com/petelomax/highlightjs-phix

File: CHANGES.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
 New Grammars:
 
 - added 3rd party Lang grammar to SUPPORTED_LANGUAGES [AdamRaichu][]
+- added 3rd party Phix grammar to SUPPORTED_LANGUAGES [PeteLomax][]
+- added 3rd party C3 grammar to SUPPORTED_LANGUAGES [aliaegik][]
```
Comment (joshgoebel) [inline CHANGES.md:7]: Is this unintended?

Comment (petelomax) [reply CHANGES.md:7]: No. Github Desktop insists I did that 13 days ago but it's far more likely something I tried 13 _months_ ago.

**Comment by axtens** [comment]
So does mean that euphoria's config.json can be changed to 
```
  "online_editor": {
    "indent_style": "space",
    "indent_size": 2,
    "highlightjs_language": "phix"
  },
```
?

File: CHANGES.md (hunk 1 above)
Comment (joshgoebel) [inline CHANGES.md:7]:
```suggestion
```

**Review by joshgoebel — APPROVED** [review]

**Review by joshgoebel — APPROVED** [review]

**Pull request merged**